server/shovel-config.ts

```diff
@@ -1,69 +1,95 @@
 import { makeConfig, toJSON } from '@indexsupply/shovel-config'
-import type { Source, Table, Integration } from '@indexsupply/shovel-config'
-import {
-  Problems,
-  Bodies,
-  Solver,
-  Dust,
-  ProblemMetadata,
-  BodyMetadata
-} from './contracts'
+import type {
+  Source,
+  Table,
+  Integration,
+  PGColumnType,
+  IndexStatment
+} from '@indexsupply/shovel-config'
+import { Problems, Bodies, Solver } from './contracts'
 import { ethers } from 'ethers'
+import { camelToSnakeCase } from './src/util'
 
-const mainnet: Source = {
+export type Chain = 'mainnet' | 'sepolia' | 'garnet' | 'base_sepolia'
+type KnownSource = Source & { name: Chain }
+
+const mainnet: KnownSource = {
   name: 'mainnet',
   chain_id: 1,
   url: 'https://ethereum-rpc.publicnode.com'
 }
 
-const sepolia: Source = {
+const sepolia: KnownSource = {
   name: 'sepolia',
   chain_id: 11155111,
-  url: 'https://rpc2.sepolia.org', // 'https://ethereum-sepolia-rpc.publicnode.com' rate limited
+  url: 'https://rpc2.sepolia.org',
   batch_size: 1000,
   concurrency: 1
 }
 
-const network = sepolia.chain_id
-const contracts = Object.fromEntries(
-  [Problems, Bodies, Solver, Dust, ProblemMetadata, BodyMetadata].map(
-    (contract) => {
-      const abi = contract.abi.abi
-      return [
-        contract.abi.contractName,
-        new ethers.Contract(contract.networks[network].address, abi)
-      ]
-    }
-  )
-)
+const garnet: KnownSource = {
+  name: 'garnet',
+  chain_id: 17069,
+  url: 'https://rpc.garnetchain.com',
+  batch_size: 1000,
+  concurrency: 1
+}
+
+const baseSepolia: KnownSource = {
+  name: 'base_sepolia',
+  chain_id: 84532,
+  url: 'https://base-sepolia.infura.io/v3/049a0c3141de47f1afcbc640911be2a7',
+  batch_size: 1000,
+  concurrency: 1
+}
 
-const solTypeToPgType = {
+const solTypeToPgType: Record<string, PGColumnType> = {
   address: 'bytea',
   uint256: 'numeric',
   bytes32: 'bytea',
   int: 'int',
   bool: 'bool'
 }
 
-function camelToSnakeCase(str: string) {
-  return str.replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`)
+const STARTING_BLOCK = {
+  // mainnet: BigInt('2067803')
+  sepolia: BigInt('5716600'),
```

Comment:
maybe these as env vars too? Otherwise we could write a script that queries for them on launch and then we don't need to maintain?

Reply:
don't think there's any advantage to an env var for this since it's public and dependent on another hardcoded value (contract address).

would def like to pull it automatically though, i'll look at that